types/aws-lambda/test/dynamodb-tests.ts

```diff
@@ -120,4 +120,10 @@ const streamHandlerWithResponse: DynamoDBStreamHandler = async (event, context,
     return result;
 };
 
-const streamHandlerNoResponse: DynamoDBStreamHandler = async (event, context, callback) => {};
+const streamHandlerNoResponse: DynamoDBStreamHandler = async (event, context, callback) => {
+    const records = event.Records;
+
+    const id = records[0].dynamodb?.Keys?.Id.N;
+    const message = records[0].dynamodb?.NewImage?.Message.S;
+    const $unknown = records[0].dynamodb?.NewImage?.Message.$unknown;
```

Comment:
That's a valid concern. I added a test for this and noticed that the binary types `AttributeValue.B` and `AttributeValue.BS` have different definitions between v2 and v3:

v2:
```ts
export interface AttributeValue {
    B?: string | undefined;
}
```
v3: 
```ts 
interface BMember {
    B: Uint8Array;
}
```

This causes a type mistmatch:

![2023-01-08_09-20-33](https://user-images.githubusercontent.com/950244/211186881-bc339a76-851d-4a66-a6fa-e51de875cc79.jpg)


That's unfortunate, because it would be a breaking change even though I don't think many people are working the DynamoDB native values directly. Usually, you use `unmarshall` to convert the DDB values into native JS values:

```
/**
 * Convert a DynamoDB record into a JavaScript object.
 *
 * @param {any} data - The DynamoDB record
 * @param {unmarshallOptions} options - An optional configuration object for `unmarshall`
 */
export const unmarshall = (
  data: Record<string, AttributeValue>,
  options?: unmarshallOptions
): Record<string, NativeAttributeValue> =>
  convertToNative({ M: data }, options) as Record<string, NativeAttributeValue>;
```
https://github.com/aws/aws-sdk-js-v3/blob/df1304fadd3bb61dff4a1483af99332630b5c9de/packages/util-dynamodb/src/unmarshall.ts#L17-L27

Reply:
The correct type is what you get from lambda directly, which will always be straight out of `JSON.parse`, and thus won't include `Uint8Array`. In short, the SDK (at least v3), unlike lambda, does seem to do some custom marshaling independent of the native dynamodb/JSON conversion (eg S instead of just a string), which means you can't directly interoperate, it seems.

The stricter type does still seem valid after correcting to what you get, even if not that valuable given its use on the input side.

Reply:
Okay, that means we should probably leave it as it is.

Reply:
@zirkelc does this mean you think this should be closed?